src/data/navProvider.tsx

```diff
@@ -0,0 +1,105 @@
+import React, {
+  createContext,
+  useContext,
+  useState,
+  ReactNode,
+  FC,
+  useEffect,
+} from "react";
+import { SectionName } from "../models/menu";
+
+const initialNavContext = {
+  updateHomePosition: () => {},
+  updateProjectsPosition: () => {},
+  updateAboutPosition: () => {},
+  updateContactPosition: () => {},
+  activeSection: "Home",
+} as unknown as ValueProp;
+
+export const NavContext = createContext<ValueProp>(initialNavContext);
+
+interface ValueProp {
+  updateHomePosition: (position: number) => void;
+  updateProjectsPosition: (position: number) => void;
+  updateAboutPosition: (position: number) => void;
+  updateContactPosition: (position: number) => void;
+  activeSection: SectionName;
+}
+
+interface NavProviderProps {
+  children?: ReactNode;
+}
+
+export const useNavContext = () => {
+  return useContext(NavContext);
+};
+
+export const NavProvider: FC<NavProviderProps> = ({ children }) => {
+  const [scrollPosition, setScrollPosition] = useState(0);
+  const [homePosition, setHomePosition] = useState(0);
+  const [projectsPosition, setProjectsPosition] = useState(0);
+  const [aboutPosition, setAboutPosition] = useState(0);
+  const [contactPosition, setContactPosition] = useState(0);
+  const [activeSection, setActiveSection] = useState<SectionName>("Home");
+
+
+  const handleScroll = () => {
+    const position = window.scrollY + 80; // (80px fixed menu height)
```

Comment:
80; // (80px fixed menu height) - i would turn it into constants with a descrptive name instead of adding a comment